addons/mail/static/src/models/attachment_image.js

```diff
@@ -8,6 +8,18 @@ registerModel({
     name: 'AttachmentImage',
     identifyingFields: ['attachmentList', 'attachment'],
     recordMethods: {
+        /**
+         * Called when clicking on download icon.
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickDownload(ev) {
+            ev.stopPropagation(); // prevents from opening viewer
```

Comment:
Please don't use `stropPropagation()`, and instead `markEventAsHandled`.
If you want to prevent opening viewer, you should modify handler method of opening viewer to pre-emptively return if it's handled by click download.

The reason we prefer marking event is because it's much easier for traceability of business logic. `stropPropagation` is too strong and is very hard to trace while debugging, because `stropPropagation` kills the event as if it never existed for the following process of dispatching event (i.e. rest of capturing and bubbling phase).